specification/hugr.md

```diff
@@ -423,15 +423,86 @@ inputs of successor `i`.
 Some normalizations are possible:
 
   - If the entry node has no predecessors (i.e. is not a loop header),
-    then its contents can be moved outside the ??-node into a containing
+    then its contents can be moved outside the CFG node into a containing
     DSG.
+  - If the entry node has only one successor and that successor is the
+    exit node, the CFG node itself can be removed.
 
-  - If the entry node’s has only one successor and that successor is the
-    exit node, the CFG-node itself can be removed
+The CFG in the example below has three inputs: one (call it `v`) of type "P"
+(not specified, but with a conversion to boolean represented by "P?"), one of
+type "qubit" and one (call it `t`) of type "angle".
 
-**Example CFG (TODO update w/ Sum types)**
+The CFG has the effect of performing an `Rz` rotation on the qubit with angle
+`x`. where `x` is the constant `C` if `v` and `H(v)` are both true and `G(F(t))`
+otherwise. (`H` is a function from type "P" to type "P" and `F` and `G` are
+functions from type "angle" to type "angle".)
 
-<img src="attachments/2647818241/2647818461.png" width="768px">
+The `DFB` nodes are labelled `Entry` and `BB1` to `BB4`. Note that the first
+output of each of these is a sum type, whose arity is the number of outgoing
+control edges; the remaining outputs are those that are passed to all
+succeeding nodes.
+
+The three nodes labelled "..." are simply generating a predicate with one empty
+value to pass to the Output node.
+
+```mermaid
+flowchart
+    subgraph CFG
+        direction TB
+        subgraph Entry
+            direction TB
+            EntryIn["Input"] -- "angle" --> F
+            EntryIn -- "P" --> Entry_["P?"]
+            Entry_ -- "[()|(P)]" --> EntryOut["Output"]
+            F -- "angle" --> EntryOut
+            EntryIn -- "qubit" --> EntryOut
+        end
+        subgraph BB1
+            direction TB
+            BB1In["Input"] -- "angle" --> G
+            BB1In -- "(Order)" --> BB1_["..."]
```

Comment:
I was going to suggest `BB1In -.-> BB1_["..."]` i.e. use dotted edges for Order....but then I see you are using them for nonlocal edges. Hmmm. There are not many edge styles available :-(. How about making the nonlocal edges thick to emphasize them, as that diagram is specifically in the section on nonlocal edges? :)